src/core/utils/fileutils.cpp

```diff
@@ -845,3 +847,75 @@ bool FileUtils::unzip( const QString &zipFilename, const QString &dir, QStringLi
 
   return true;
 }
+
+bool FileUtils::isDeletable( const QString &filePath )
+{
+  const QFileInfo fileInfo( filePath );
+  if ( !fileInfo.exists() || !fileInfo.isFile() )
+    return false;
+
+  const QString canonicalFilePath = fileInfo.canonicalFilePath();
+  const QString appDataDir = QFileInfo( PlatformUtilities::instance()->applicationDirectory() ).canonicalFilePath();
+  const QString cloudDataDir = QFileInfo( QFieldCloudUtils::localCloudDirectory() ).canonicalFilePath();
+  const QStringList extraDirs = PlatformUtilities::instance()->additionalApplicationDirectories();
+
+  const bool isInsideAppDataDir = !appDataDir.isEmpty() && canonicalFilePath.startsWith( appDataDir );
+  const bool isInsideCloudDataDir = !cloudDataDir.isEmpty() && canonicalFilePath.startsWith( cloudDataDir );
+  const bool isInsideExtraDir = std::any_of( extraDirs.begin(), extraDirs.end(), [&canonicalFilePath]( const QString &dir ) {
+    if ( dir.isEmpty() )
+      return false;
+
+    const QString canonicalDir = QFileInfo( dir ).canonicalFilePath();
+    return !canonicalDir.isEmpty() && canonicalFilePath.startsWith( canonicalDir );
+  } );
+
+  if ( !( isInsideAppDataDir || isInsideCloudDataDir || isInsideExtraDir ) )
+    return false;
```

Comment:
I think it'd be nicer to have a QStringList of directories with which you append app, cloud, and extra directories, then do _one_ std::any_of against that list.